File: app/src/main/java/com/github/alexeygorovoy/moxy_template/dagger/demo/heroes/HeroesModule.java
package com.github.alexeygorovoy.moxy_template.dagger.demo.heroes;

import com.github.alexeygorovoy.moxy_template.api.HeroApi;
import com.github.alexeygorovoy.moxy_template.ui.demo.heroes.presenter.HeroesListPresenter;
import com.github.alexeygorovoy.moxy_template.utils.rx.RxSchedulers;

import dagger.Module;
import dagger.Provides;


@Module
public class HeroesModule {

    @Provides
    HeroesListPresenter provideHeroesListPresenter(HeroApi heroApi, RxSchedulers rxSchedulers) {
        return new HeroesListPresenter(heroApi, rxSchedulers);
    }


}

File: app/src/main/java/com/github/alexeygorovoy/moxy_template/dagger/demo/heroes/HeroesModule.kt
package com.github.alexeygorovoy.moxy_template.dagger.demo.heroes

import com.github.alexeygorovoy.moxy_template.api.HeroApi
import com.github.alexeygorovoy.moxy_template.ui.demo.heroes.presenter.HeroesListPresenter
import com.github.alexeygorovoy.moxy_template.utils.rx.RxSchedulers

import dagger.Module
import dagger.Provides

@Module
class HeroesModule {

    @Provides
    internal fun provideHeroesListPresenter(heroApi: HeroApi, rxSchedulers: RxSchedulers) = HeroesListPresenter(heroApi, rxSchedulers)
}
